src/dataEntities/appointment.ts

```diff
@@ -60,11 +61,15 @@ export abstract class EthereumAppointment implements IEthereumAppointment {
     get passedInspection() {
         return this.mPassedInspection;
     }
-    setInspectionResult(passed, startBlock) {
+    setInspectionResult(passed: boolean, startBlock: number | null) {
         this.mPassedInspection = passed;
         if (passed) {
-            this.mStartBlock = startBlock;
-            this.mEndBlock = startBlock + this.expiryPeriod;
+            if (startBlock === null) {
+                throw new ApplicationError("startBlock must be provided on a passed inspection.");
+            } else {
+                this.mStartBlock = startBlock;
+                this.mEndBlock = startBlock + this.expiryPeriod;
+            }
```

Comment:
Here I'm assuming that startBlock is ignored if `passed === false`, but it must not be `null` otherwise.

Reply:
This method seems a bit outdated. It seems we never used the passed: false variant. So lets refactor this method, call it passInspection(startBlock: number), and not allow startBlock to be null.

Reply:
As a side note, we have an ArgumentError error that you can throw when checking arguments.

Reply:
Refactored as suggested.